x-pack/plugins/security_solution/server/lib/risk_engine/tasks/risk_scoring_task.ts

```diff
@@ -153,101 +163,132 @@ export const runTask = async ({
-    dataViewId,
-  });
-  const identifierTypes: IdentifierType[] = configuredIdentifierType
-    ? [configuredIdentifierType]
-    : [RiskScoreEntity.host, RiskScoreEntity.user];
-
-  await asyncForEach(identifierTypes, async (identifierType) => {
-    let isWorkComplete = false;
-    let afterKeys: AfterKeys = {};
-    while (!isWorkComplete) {
-      const result = await riskScoreService.calculateAndPersistScores({
-        afterKeys,
-        index,
-        filter,
-        identifierType,
-        pageSize,
-        range,
-        runtimeMappings,
-        weights: [],
-      });
+    const configuration = await riskScoreService.getConfiguration();
+    if (configuration == null) {
+      log(
+        'Risk engine configuration not found; exiting task. Please reinitialize the risk engine and try again'
+      );
+      return { state: updatedState };
+    }
 
-      isWorkComplete = isRiskScoreCalculationComplete(result);
-      afterKeys = result.after_keys;
-      scoresWritten += result.scores_written;
+    const {
+      dataViewId,
+      enabled,
+      filter,
+      identifierType: configuredIdentifierType,
+      range: configuredRange,
+      pageSize,
+    } = configuration;
+    if (!enabled) {
+      log('risk engine is not enabled, exiting task');
+      return { state: updatedState };
     }
-  });
 
-  updatedState.scoresWritten = scoresWritten;
+    const range = convertRangeToISO(configuredRange);
+    const { index, runtimeMappings } = await riskScoreService.getRiskInputsIndex({
+      dataViewId,
+    });
+    const identifierTypes: IdentifierType[] = configuredIdentifierType
+      ? [configuredIdentifierType]
+      : [RiskScoreEntity.host, RiskScoreEntity.user];
 
-  log('task run completed');
-  return {
-    state: updatedState,
-  };
+    await asyncForEach(identifierTypes, async (identifierType) => {
+      let isWorkComplete = false;
+      let afterKeys: AfterKeys = {};
+      while (!isWorkComplete) {
+        const result = await riskScoreService.calculateAndPersistScores({
+          afterKeys,
+          index,
+          filter,
+          identifierType,
+          pageSize,
+          range,
+          runtimeMappings,
+          weights: [],
+        });
+
+        isWorkComplete = isRiskScoreCalculationComplete(result);
+        afterKeys = result.after_keys;
+        scoresWritten += result.scores_written;
+      }
+    });
+
+    updatedState.scoresWritten = scoresWritten;
+
+    const taskCompletionTime = moment().utc().toISOString();
+
+    const taskDurationInSeconds = moment(taskCompletionTime).diff(moment(taskStartTime), 'seconds');
+
+    const telemetryEvent = {
+      scoresWritten,
+      taskDurationInSeconds,
+      executionDurationExceededInterval: isExecutionDurationExceededInterval(
+        taskInstance?.schedule?.interval,
+        taskDurationInSeconds
+      ),
+    };
+
+    telemetry.reportEvent(RISK_SCORE_EXECUTION_SUCCESS_EVENT.eventType, telemetryEvent);
```

Comment:
My only comment is why not centralize this with the rest of the telemetry tasks in `/server/lib/telemetry`, but it's up to you.